File: AwsSamples/CdkSamBlazorAspNetCoreDynamoDb/ServerlessApi/Controllers/WeatherForecastController.cs
using Common.Models;
using Microsoft.AspNetCore.Mvc;

namespace ServerlessApi.Controllers;

[ApiController]
[Route("[controller]")]
public class WeatherForecastController : ControllerBase
{
    [HttpGet]
    public List<WeatherForecast> Get()
    {
        return new List<WeatherForecast>
        {
            new()
            {
                Date = new DateTime(2018, 5, 6),
                TemperatureC = 1,
                Summary = "Freezing"
            },
            new()
            {
                Date = new DateTime(2018, 5, 7),
                TemperatureC = 14,
                Summary = "Bracing"
            },
            new()
            {
                Date = new DateTime(2018, 5, 8),
                TemperatureC = -13,
                Summary = "Freezing"
            },
            new()
            {
                Date = new DateTime(2018, 5, 9),
                TemperatureC = -16,
                Summary = "Balmy"
            },
            new()
            {
                Date = new DateTime(2018, 5, 10),
                TemperatureC = -2,
                Summary = "Chilly"
            }
        };
    }
}

File: FSharp/Api/Controllers/WeatherForecastController.fs
namespace Api.Controllers

open System
open Microsoft.AspNetCore.Mvc
open Microsoft.Extensions.Logging

type WeatherForecast =
  { Date: DateTime
    TemperatureC: int
    Summary: string }

  member this.TemperatureF = 32.0 + (float this.TemperatureC / 0.5556)

[<ApiController>]
[<Route("api/[controller]")>]
type WeatherForecastController(logger: ILogger<WeatherForecastController>) =
  inherit ControllerBase()

  let summaries =
    [| "Freezing"
       "Bracing"
       "Chilly"
       "Cool"
       "Mild"
       "Warm"
       "Balmy"
       "Hot"
       "Sweltering"
       "Scorching" |]

  [<HttpGet>]
  member _.Get() =
    logger.LogInformation("Getting weather forecast")
    let rng = Random()

    [| for index in 0..4 ->
         { Date = DateTime.Now.AddDays(float index)
           TemperatureC = rng.Next(-20, 55)
           Summary = summaries.[rng.Next(summaries.Length)] } |]
